swift/Arrow/Tests/ArrowTests/CodableTests.swift

```diff
@@ -134,7 +136,26 @@ final class CodableTests: XCTestCase {
             let testData = try decoder.decode(Int8?.self)
             for index in 0..<testData.count {
                 let val: Int8? = testData[index]
-                if val != nil {
+                XCTAssertEqual(val!, Int8(index + 10))
+            }
+        case .failure(let err):
+            throw err
+        }
+
+        let int8WNilBuilder: NumberArrayBuilder<Int8> = try ArrowArrayBuilders.loadNumberArrayBuilder()
+        int8WNilBuilder.append(10, nil, 12, nil)
+        let resultWNil = RecordBatch.Builder()
+            .addColumn("propInt8", arrowArray: try int8WNilBuilder.toHolder())
+            .finish()
+        switch resultWNil {
+        case .success(let rb):
+            let decoder = ArrowDecoder(rb)
+            let testData = try decoder.decode(Int8?.self)
+            for index in 0..<testData.count {
+                let val: Int8? = testData[index]
+                if index % 2 == 1 {
+                    XCTAssertNil(val)
+                } else {
```

Comment:
How about creating a new separated test for this case?
`testArrowSingleDecoderWithoutNull()` and `testArrowSingleDecoderWithNull()`?